### PR TITLE
Add 1-20 patches

### PR DESCRIPTION
**Pull request by jyotimahapatra**
*Description of changes:*
This PR attempts to bring in the patches for 1.20 release.

k8s tag: 1.20.4
All the current 1.19 patches were applied on the 1.20 tag. Conflicts were resolved manually to create the new patches. 
https://github.com/aws/eks-distro/blob/main/projects/kubernetes/kubernetes/1-19/patches/0007-EKS-PATCH-AWS-cloudprovider-allow-nlb-ip-and-externa.patch and https://github.com/aws/eks-distro/blob/main/projects/kubernetes/kubernetes/1-19/patches/0008-EKS-PATCH-Update-aws-sdk-go-to-v1.34.24.patch have been merged upstream. They have been removed from the patch list.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/kubernetes/kubernetes/1-20/checksums [hunk 1]
```diff
@@ -0,0 +1,19 @@
+1526a6d976277207ed4c3398b457031aa3908f671c1871882eaee7a64098a2b6  _output/1-19/bin/windows/amd64/kube-proxy.exe
```
Comment (jyotimahapatra) [inline projects/kubernetes/kubernetes/1-20/checksums:1]: how are these checksums calculated?
Would appreciate instructions in the readme.md about the process for updating relevant files.

**Comment by jyotimahapatra** [comment]
@kschumy Could you help review this one? Apart from the patches, I took my best judgement to update the other relevant files, but i'm sure there are some gaps. Could you help identify what else is missing?

**Comment by kschumy** [comment]
This is great! Thanks for getting a jump on it. I'll look over this in the morning

**Comment by jaxesn** [comment]
This is awesome!  Here is a rough outline on the steps.  @kschumy maybe you can make sure this gets into the readme in some form?

1. Create pre + post submit jobs
   * Create presubmit from [previous](https://github.com/aws/eks-distro-prow-jobs/blob/main/jobs/aws/eks-distro/kubernetes-1-19-presubmits.yaml) job, changing preview minor version (1-19) to new version (1-20).  The presubmit job ones before allowing merges to main and will build Kubernetes and run the upstream unit tests to validate the patches do not break any known functionality.
   * Create postsubmit from [previous](https://github.com/aws/eks-distro-prow-jobs/blob/main/jobs/aws/eks-distro/dev-release-1-19-postsubmits.yaml).  The postsubmit job runs after merging to master and will create a kubernetes cluster using kops and the new built version and then run the upstream conformance tests.
2. Create boilerplate files for new kubernetes version
   * echo 'v1.XX.X' > 1-XX/GIT_TAG
   * echo '0' > 1-XX/RELEASE
   * echo '0' > 1-XX/RELEASE-DEV
   * echo '0' > 1-XX/RELEASE-PUBLIC
3. Validate build with new patches, can be run on Mac or Linux machine
   * RELEASE_BRANCH=1-XX make binaries
4. Generate KUBE_GIT_VERSION_FILE file for new version
   * RELEASE_BRANCH=1-XX make update-version 
5. Generate checksums for new version
   * RELEASE_BRANCH=1-XX make update-checksums 
6. Run unit tests
   *  RELEASE_BRANCH=1-XX make test

The intention was to make steps 3-6 easier by introducing make [update-kubernetes-version  ](https://github.com/aws/eks-distro/blob/main/Makefile#L88) via the root Makefile.  That uses a container based on the builder-base we use in CI to build/generate/test the new kubernetes version.  It is also the design that our builds are reproducible, which is why we check in the checksums and these checksums need to be generated on a linux based machine vs a mac.  However, currently our builds produce different checksums so this isn't a huge concern atm.  The builder-base image is not in a public ecr repo atm so using that approach instead steps 3-6 only works for someone on our team with access to the private image.  We can def make these a public going forward.  For future reference, here are the steps I used on my mac the container based approach:

1. From root of eks-distro repo
   * RELEASE_BRANCH=1-XX make update-kubernetes-version  
2. Run unit tests
   * docker run -it -e RELEASE_BRANCH=1-XX eks-distro-local-kubernetes-update:v1.XX.X-0 make test

I ran the above off your branch and these are two files it produced, if you want them.

KUBE_GIT_VERSION_FILE 
```
KUBE_GIT_COMMIT='e87da0bd6e03ec3fea7933c4b5263d151aafd07c'
KUBE_GIT_VERSION='v1.20.4-eks-1-20-0'
KUBE_GIT_MAJOR='1'
KUBE_GIT_MINOR='20'
SOURCE_DATE_EPOCH='1613664121'
KUBE_GIT_TREE_STATE='archive'
``` 
checksums
```
3be8a162f1c3a12363cef8bf44ce11b0eb66d47fd7a042af062f69b777b54d79  _output/1-20/bin/windows/amd64/kube-proxy.exe
6427b5a2f49290e6f9efc36dd332949cc3db2bc08b1a38a702f8bf2714754be8  _output/1-20/bin/windows/amd64/kubectl.exe
adaf45e56db901f17970a16ed3a413eac594b65645f874b9fd2347ad1f371699  _output/1-20/bin/windows/amd64/kubelet.exe
b8428e1551f597b91f533f8e462639c3086e7a5557e8cf812fc9bec94237d1b5  _output/1-20/bin/windows/amd64/kubeadm.exe
dfcf74a93157200e26e1ad72c8e09b71d518e2d7b66b7c91f1f026af597448ba  _output/1-20/bin/linux/amd64/kubeadm
33f87b6580e5795dd35dd1a4f2dfaaf97b66587859af502bc1d150640e9e66b7  _output/1-20/bin/linux/amd64/kube-controller-manager
d72db0e0a46c4935039070a9ca132b5035949a2389dea5d083fb030924268d22  _output/1-20/bin/linux/amd64/kubectl
d27ec7481a941fb1dfac656a5772634247139899448a4e82ce2fe1f608bb9a3b  _output/1-20/bin/linux/amd64/kube-apiserver
fb21f6a49bbe146a76b5cf4ea89769d4befcd423995a6b3520fe45977099a21d  _output/1-20/bin/linux/amd64/kubelet
a1b56cdf9310eec6e7776352d99df818936c422d95fe9f908abfdca35a3888e1  _output/1-20/bin/linux/amd64/kube-proxy
8cfefdef417cf88e052de3babcf70da5668b70751ddad356a75d604a03de2d2a  _output/1-20/bin/linux/amd64/kube-scheduler
8b18fdff7ea0b85c8a9de22fe8790d58c2a33f3f790346c22aee1b4a06893110  _output/1-20/bin/linux/arm64/kubeadm
d6d2e1023075214716591503d87ab840deb73a82d9418e08b9f99cad4f70df2d  _output/1-20/bin/linux/arm64/kube-controller-manager
e8ffb375fd3763ce51cc00b8dcf97778239b5e4317058dd98f1f6806cc4c97d2  _output/1-20/bin/linux/arm64/kubectl
c39a7b6b1ba47bf9b20afdb26d21729d285da67aec05238383b4c546ad861316  _output/1-20/bin/linux/arm64/kube-apiserver
bdd2fed4c213a8bfc6a856a47af4ed3025da363f09bfaf0436b7bd47cdbe82f8  _output/1-20/bin/linux/arm64/kubelet
e2f755bf67ffce1f71f8d82002b21239d3de21322f2ab36e55b5fb68a540785f  _output/1-20/bin/linux/arm64/kube-proxy
6fe2b4e9575bcb2162be1ed235de4f0934b06f4abd31ea37186b8c6d8122b00d  _output/1-20/bin/linux/arm64/kube-scheduler
118635669993828e864515e8cc97046c0be9965cc057a3431a99d62ae13d898d  _output/1-20/bin/darwin/amd64/kubectl
```

**Comment by jyotimahapatra** [comment]
@jaxesn Thanks for the instructions and the checksums. I've updated them on the PR now.
I noticed that there's no `make test` available. 
```
ubuntu@ip-xx:~/src/eks-distro$ RELEASE_BRANCH=1-20 make
binaries                   docker-push                release                    upload
build                      makes                      setup
clean                      postsubmit-build           tag
docker                     postsubmit-conformance     update-kubernetes-version
```

**Comment by jaxesn** [comment]
Sorry for just now responding, the make test is on the Makefile in the projects/kubernetes/kubernetes folder vs the root directory. I ran them locally from your branch and everything passed except for some of the common flakes which are also flaky upstream.

**Comment by jyotimahapatra** [comment]
@jaxesn Thanks...i added a missing patch .
It was present in 1.18 but missing in 1.19 ....does that lead to regeneration of checksum?

**Comment by jyotimahapatra** [comment]
@jaxesn The 1.20 jobs are failing with error `panic: no space left on device` . Appreciate any help to look into it.

**Comment by kschumy** [comment]
> @kschumy can you check the notes and see why that patch was excluded in 1.19?

This was the one that was accidentally left off. We need to add it back in for 1.19. Jyoti is going to add it to 1.20

File: projects/kubernetes/kubernetes/1-20/patches/0008-EKS-PATCH-additional-subnet-configuration-for-AWS-EL.patch [hunk 2]
```diff
@@ -0,0 +1,695 @@
+From 039fbec8dd1d9d496ba86de1a0047dcd45d5e53f Mon Sep 17 00:00:00 2001
+From: Jyoti Mahapatra <jyotima@amazon.com>
+Date: Tue, 9 Mar 2021 05:43:04 +0000
+Subject: [PATCH] --EKS-PATCH-- additional subnet configuration for AWS ELB
+
```
Comment (kschumy) [inline projects/kubernetes/kubernetes/1-20/patches/0008-EKS-PATCH-additional-subnet-configuration-for-AWS-EL.patch:5]: To keep consistent with how [this patch was in the last release](https://github.com/kschumy/eks-distro/blob/main/projects/kubernetes/kubernetes/1-19/patches/0010-EKS-PATCH-additional-subnet-configuration-for-AWS-el.patch#L6-L9), can we add the following as a description? 

```
* support subnets without cluster tag for auto-discovery
* add support for manual subnet configuration via service annotation

ref: https://github.com/kubernetes/kubernetes/pull/97431
```

Comment (jyotimahapatra) [reply projects/kubernetes/kubernetes/1-20/patches/0008-EKS-PATCH-additional-subnet-configuration-for-AWS-EL.patch:5]: done

**Comment by jyotimahapatra** [comment]
@jaxesn Thank you! The new error in 1-20 is
```
#3 [internal] load metadata for public.ecr.aws/h1r8a7l5/kubernetes/go-runne...
#3 ERROR: public.ecr.aws/h1r8a7l5/kubernetes/go-runner:v0.4.2-eks-1-20-0: not found
------
 > [internal] load metadata for public.ecr.aws/h1r8a7l5/kubernetes/go-runner:v0.4.2-eks-1-20-0:
------
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: public.ecr.aws/h1r8a7l5/kubernetes/go-runner:v0.4.2-eks-1-20-0: not found
```

LMK If there are instructions for how to create the new go-runner tag `go-runner:v0.4.2-eks-1-20-0`

**Comment by TerryHowe** [comment]
go-runner not being there is a chicken or egg type problem. I'm looking at it. That would be triggered by the postsubmit.

**Comment by jyotimahapatra** [comment]
@TerryHowe My bad..i've fixed the patch now

**Comment by jyotimahapatra** [comment]
Its not failing again with `public.ecr.aws/h1r8a7l5/kubernetes/go-runner:v0.4.2-eks-1-20-0: not found`

**Comment by TerryHowe** [comment]
That one change didn't trigger a release build and 1.18 and 1.19 failed for no space on device. Reverted that one change.

File: projects/kubernetes/kubernetes/1-20/patches/0009-EKS-PATCH-aws_credentials-update-ecr-url-validation-.patch [hunk 3]
```diff
@@ -0,0 +1,46 @@
+From 80a1017709a8bf689d4d75b8df1970198d35ccb8 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?chrisDallas=20=E2=98=95=20=F0=9F=8C=A7=EF=B8=8F=20?=
+ =?UTF-8?q?=E2=98=82?= <cdalla@amazon.com>
+Date: Thu, 8 Oct 2020 11:58:58 -0700
+Subject: --EKS-PATCH-- (aws_credentials): update ecr url validation
+ regex
+
+Signed-off-by: Jackson West <jgw@amazon.com>
```
Comment (kschumy) [inline projects/kubernetes/kubernetes/1-20/patches/0009-EKS-PATCH-aws_credentials-update-ecr-url-validation-.patch:8]: Why is Jackson the sign-off for this one?

**Comment by TerryHowe** [comment]
PTAL https://github.com/aws/eks-distro-prow-jobs/pull/125 I'm hoping that will resolve this one way or the other

File: projects/kubernetes/kubernetes/1-20/patches/0009-EKS-PATCH-aws_credentials-update-ecr-url-validation-.patch (hunk 3 above)
Comment (TerryHowe) [reply projects/kubernetes/kubernetes/1-20/patches/0009-EKS-PATCH-aws_credentials-update-ecr-url-validation-.patch:8]:
```suggestion
Signed-off-by: Jyoti Mahapatra <jyotima@amazon.com>
```

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Comment by TerryHowe** [comment]
Should merge if tests pass

**Pull request merged**